slasher/db/attester_slashings.go

```diff
@@ -0,0 +1,179 @@
+package db
+
+import (
+	"bytes"
+
+	"github.com/prysmaticlabs/go-ssz"
+
+	"github.com/boltdb/bolt"
+	"github.com/gogo/protobuf/proto"
+	"github.com/pkg/errors"
+	ethpb "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
+)
+
+func createAttesterSlashing(enc []byte) (*ethpb.AttesterSlashing, error) {
+	protoSlashing := &ethpb.AttesterSlashing{}
+
+	err := proto.Unmarshal(enc, protoSlashing)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to unmarshal encoding")
+	}
+	return protoSlashing, nil
+}
+
+func toAttesterSlashings(encoded [][]byte) ([]*ethpb.AttesterSlashing, error) {
+	var attesterSlashings []*ethpb.AttesterSlashing
+	for _, enc := range encoded {
+		ps, err := createAttesterSlashing(enc)
+		if err != nil {
+			return nil, err
+		}
+		attesterSlashings = append(attesterSlashings, ps)
+	}
+	return attesterSlashings, nil
+}
+
+// AttesterSlashings accepts a status and returns all slashings with this status.
+// returns empty []*ethpb.AttesterSlashing if no slashing has been found with this status.
+func (db *Store) AttesterSlashings(status SlashingStatus) ([]*ethpb.AttesterSlashing, error) {
+	encoded := make([][]byte, 0)
+	err := db.view(func(tx *bolt.Tx) error {
+		c := tx.Bucket(slashingBucket).Cursor()
+		prefix := encodeStatusType(status, SlashingType(Attestation))
+		for k, v := c.Seek(prefix); k != nil && bytes.HasPrefix(k, prefix); k, v = c.Next() {
+			encoded = append(encoded, v)
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+	return toAttesterSlashings(encoded)
+}
+
+// AttestingSlashingsByStatus return all attestation slashing proofs with certain status.
+func (db *Store) AttestingSlashingsByStatus(status SlashingStatus) ([]*ethpb.AttesterSlashing, error) {
```

Comment:
what is the difference between this method and the above one ? they are exactly the same.

Reply:
thanks, removed